### PR TITLE
Fix button clicks blocklets

### PR DESCRIPTION
**Pull request by moritzheiber**
- Also fix network bandwidth calculation/shellcheck errors
  in net-traffic script

Fixes https://github.com/regolith-linux/regolith-desktop/issues/336

### CODE REVIEW

**Comment by cheginit** [comment]
We're planning to add a floating window class to the i3 config file called `floating_window`. So for the button section, you can add `class=floating_window` flag to `gnome-control-center` to make the window floating.

See this [PR](https://github.com/regolith-linux/regolith-i3-gaps-config/pull/16/commits/bdf71bc31c132de029fcca71dbe5c2caf4406066).

**Comment by moritzheiber** [comment]
@cheginit For me the window is floating when run from the blocklet, though, likely because [there's an option which applies to all GNOME Control Center windows](https://github.com/regolith-linux/regolith-i3-gaps-config/blob/master/config#L332).. is setting the explicit class even necessary then?

**Comment by cheginit** [comment]
We're planning to replace it with this approach to make it more general so users can make any window floating by using this flag. It's a gtk flag so all gtk apps that implement it can be made floating.

**Comment by moritzheiber** [comment]
> We're planning to replace it with this approach to make it more general so users can make any window floating by using this flag. It's a gtk flag so all gtk apps that implement it can be made floating.

Added the window class to all calls to `gnome-control-center`

File: scripts/net-traffic [hunk 1]
```diff
@@ -1,5 +1,7 @@
 #!/bin/bash
 
+set -Eeu -o pipefail
```
Comment (kgilmer) [inline scripts/net-traffic:3]: Interesting!  What's going on here?

File: scripts/net-traffic [hunk 2]
```diff
@@ -21,8 +23,22 @@ GET="cat /proc/net/dev"
...
+BUTTON=${button:-}
+DLY=${DLY:-1}
+RT=${RT:-}
+BPS=false
+UNIT=KB
 ULEV=2
 
+# Handle button event
+if [ "x${BUTTON}" == "x1" ] ; then
+    if [ "$INT_TYPE" == "W" ]; then
```
Comment (kgilmer) [inline scripts/net-traffic:35]: Nice!

File: scripts/net-traffic [hunk 3]
```diff
@@ -143,44 +154,6 @@ RX=$(echo "scale = 0; ($RX2-$RX1)/$DLY" | bc -l)
 TX=$(echo "scale = 0; ($TX2-$TX1)/$DLY" | bc -l)
 AX=$(echo "scale = 0; $RX+$TX" | bc -l)
 
-# set net unit automatically
```
Comment (kgilmer) [inline scripts/net-traffic:146]: This can be removed because it's redundant?

File: scripts/wifi [hunk 4]
```diff
@@ -1,16 +1,17 @@
...
 
+BUTTON=${button:-}
 VALUE_COLOR=${color:-$(xrescat i3xrocks.value.color "#D8DEE9")}
 LABEL_ICON=${label_icon:-$(xrescat i3xrocks.label.wifi )}
 LABEL_COLOR=${label_color:-$(xrescat i3xrocks.label.color "#7B8394")}
 VALUE_FONT=${font:-$(xrescat i3xrocks.value.font "Source Code Pro Medium 13")}
 
 ESSID=$(LC_ALL=C nmcli -t -f active,ssid dev wifi |  perl -n -e '/yes:(.*)/ && print $1')
-if [ ! -z "$ESSID" ]; then
+if [ -n "$ESSID" ]; then
```
Comment (kgilmer) [inline scripts/wifi:10]: I'm learning!

File: scripts/net-traffic (hunk 3 above)
Comment (moritzheiber) [reply scripts/net-traffic:146]: Yes, I’m getting rid of it in favor of a sane default (which also fixes a shellcheck error)

File: scripts/net-traffic (hunk 1 above)
Comment (moritzheiber) [reply scripts/net-traffic:3]: `-E` makes sure we're inheriting errors from subshells
`-e` exits the script immediately upon an error (i.e. it doesn't "bleed" its errors either into userspace or, worse, the bar)
`-u` treats unset variables as errors (i.e. you have to initialize them or the script won't run)
`-o pipefail` fails the script with the status of the last pipeline execution (and not of the overall script)

Comment (moritzheiber) [reply scripts/net-traffic:3]: Generally, these are safe to use when you're running a properly written script

File: scripts/net-traffic (hunk 3 above)
Comment (moritzheiber) [reply scripts/net-traffic:146]: We could keep both around though, if you prefer to have the code instead of the (arguable, forced) default

**Comment by jefft** [comment]
Ahem, commit  8e7daa1 left the 'x' flag on in the net-traffic script, and now my `i3-gnome-flashback.desktop` systemd log is being filled with debug logs :)

**Comment by kgilmer** [comment]
Thanks for helping us test in `unstable` @jefft :smile: .  We'll get that fixed up shortly.

**Pull request merged**